### PR TITLE
Handle special case, if users have no access to the root group.

### PR DESCRIPTION
**Pull request by aquamatthias**

### CODE REVIEW

**Comment by aquamatthias** [comment]
@gkleiman added a test case

File: src/test/scala/mesosphere/marathon/api/v2/GroupsResourceTest.scala [hunk 1]
```diff
@@ -136,6 +136,20 @@ class GroupsResourceTest extends MarathonSpec with Matchers with Mockito with Gi
...
+    Given("An unauthenticated request")
+    auth.authenticated = true
+    auth.authorized = false
+    val req = auth.request
+    groupInfo.selectGroup(any, any, any, any) returns Future.successful(None)
+
+    When(s"the root is fetched from index")
+    val root = groupsResource.root(req, embed)
+
+    Then("we receive a NotAuthenticated response")
```
Comment (gkleiman) [inline src/test/scala/mesosphere/marathon/api/v2/GroupsResourceTest.scala]: the request is successful =)

**Comment by gkleiman** [comment]
@aquamatthias: just one more little thing and then I'll hit merge! =)

**Comment by aquamatthias** [comment]
@gkleiman copy/paste. Thanks.

**Comment by gkleiman** [comment]
:ship:

**Pull request merged**